src/main/java/seedu/duke/commands/AddCommand.java

```diff
@@ -1,15 +1,169 @@
+    }
+
+    private int retrieveEventBudget(String response) {
+        int startOfEventBudget = response.indexOf("b/") + 2;
+        int endOfEventBudget = response.indexOf("/", startOfEventBudget) - 2;
+        try {
+            if (endOfEventBudget < 0) {
+                return Integer.parseInt(response.trim().substring(startOfEventBudget));
+            }
+            return Integer.parseInt(response.trim().substring(startOfEventBudget, endOfEventBudget));
+        } catch (NumberFormatException e) {
+            System.out.print("Budget needs to be an integer. ");
+            return -1;
+        }
+    }
+
+    private void prepareTask(String response) throws DukeException {
+        if (!response.contains("n/")) {
+            throw new DukeException("Please add a title for your task using 'n/<title>'. ");
+        }
+        if (!response.contains("d/")) {
+            throw new DukeException("Please add a deadline for your task using 'd/<deadline>' in the "
+                    + "format YYYY-MM-DD. ");
+        }
+
+        itemTitle = retrieveItemTitle(response);
+        if (itemTitle.equals("")) {
+            throw new DukeException("Task title cannot be empty, please re-enter your task. ");
+        }
+
+        itemDate = retrieveItemDate(response);
+        if (itemDate.equals("")) {
+            throw new DukeException("Task deadline cannot be empty, please re-enter your task. ");
+        }
+    }
+
+    private void prepareEvent(String response) throws DukeException {
+        if (!response.contains("n/")) {
+            throw new DukeException("Please add a title for your event using 'n/<title>'. ");
+        }
+        if (!response.contains("d/")) {
+            throw new DukeException("Please add a date for your event using 'd/<date>' in the "
+                    + "format YYYY-MM-DD. ");
+        }
+        if (!response.contains("v/")) {
+            throw new DukeException("Please add a venue for your event using 'v/<venue>'. ");
+        }
+        if (!response.contains("b/")) {
+            throw new DukeException("Please add a budget for your event using 'b/<budget>'. ");
+        }
+
+        itemTitle = retrieveItemTitle(response);
+        if (itemTitle.equals("")) {
+            throw new DukeException("Event title cannot be empty, please re-enter your event. ");
+        }
+
+        itemDate = retrieveItemDate(response);
+        if (itemDate.equals("")) {
+            throw new DukeException("Event date cannot be empty, please re-enter your event. ");
+        }
+
+        eventVenue = retrieveEventVenue(response);
+        if (eventVenue.equals("")) {
+            throw new DukeException("Event venue cannot be empty, please re-enter your event. ");
+        }
+
+        eventBudget = retrieveEventBudget(response);
+        if (eventBudget == -1) {
+            throw new DukeException("Please re-enter your event. ");
+        }
+    }
```

Comment:
Good extensive throwing of exceptions to catch the possible errors when adding items.
One possible suggestion could be changing `.equals("")` to `.empty()`, ie changing `itemDate.equals("")` to `itemDate.empty()` as we are dealing with `String` objects and could utilise methods from the `String` API to make the code more self-explanatory.